lib/barcode/barcode.dart

```diff
@@ -462,6 +463,113 @@ class ScanParentLocationHandler extends BarcodeScanStockLocationHandler {
 }
 
 
+/*
+ * Barcode handler class for scanning a supplier barcode to receive a part
+ *
+ * - The class can be initialized by optionally passing a valid, placed PurchaseOrder object
+ * - Expects to scan supplier barcode, possibly containing order_number and quantity
+ * - If location or quantity information wasn't provided, show a form to fill it in
+ */
+class POReceiveBarcodeHandler extends BarcodeHandler {
+
+  POReceiveBarcodeHandler({this.purchaseOrder, this.location});
+
+  InvenTreePurchaseOrder? purchaseOrder;
+  InvenTreeStockLocation? location;
+
```

Comment:
You need to add a custom `getOverlayText` method to this class, currently it just says "barcode overlay" at the bottom of the screen :)

Reply:
Done!